### PR TITLE
DietPi-Software | Add folding@home to software list

### PR DESCRIPTION
**Pull request by MichaIng**
**Status**: WIP
- [x] Software category?
- [x] DietPi-Services handling?
- [x] Allow fully automated install by providing user name, passkey and power choice via dietpi.txt?
- [x] Move data dir to dietpi_userdata?
- [x] Changelog entry

**Reference**: https://github.com/Fourdee/DietPi/issues/1985

**Commit list/description**:
+ DietPi-Software | Add folding@home to software list
  - Preconfigured autostart + DietPi team
  - User name, passkey, power choice need to be entered

### CODE REVIEW

File: dietpi/dietpi-software [hunk 1]
```diff
@@ -1514,6 +1514,24 @@ _EOF_
 	   aSOFTWARE_REQUIRES_JAVA_JRE_JDK[$index_current]=1
 			 aSOFTWARE_ONLINEDOC_URL[$index_current]='f=8&t=5&p=6202#p6202'
```
Comment (MichaIng) [inline dietpi/dietpi-software:1516]: @Fourdee 
Watching alignment of the software arrays, I guess your local code editor handled tabs differently than github (tab = 8 digits space)?
Found my notepad++ with default 2 spaces, if I remember right, so raw downloads from github looked wrong at first.

Shall we align all of them to look pretty on github... by times... ?

Comment (Fourdee) [reply dietpi/dietpi-software:1516]: @MichaIng 

I use notepad++ with default tab size of 4
image

image

> Shall we align all of them to look pretty on github... by times... ?

Yep, we probably should do that (at some point, minor). Apparently GitHub default is 8 for a tab:
https://github.com/lukechilds/github-custom-tab-size#github-custom-tab-size

File: dietpi/dietpi-software [hunk 2]
```diff
@@ -1514,6 +1514,24 @@ _EOF_
...
 
+		#------------------
+		index_current=2
+
+		aSOFTWARE_WHIP_NAME[$index_current]='Folding@Home'
+		aSOFTWARE_WHIP_DESC[$index_current]='distributed disease research project'
+		aSOFTWARE_CATEGORY_INDEX[$index_current]=6
+		aSOFTWARE_TYPE[$index_current]=0
+		aSOFTWARE_ONLINEDOC_URL[$index_current]=''
```
Comment (Fourdee) [inline dietpi/dietpi-software]: Created:
https://dietpi.com/phpbb/viewtopic.php?f=8&t=5&p=13704#p13704

Entry:
```
f=8&t=5&p=13704#p13704
```

File: dietpi/dietpi-software (hunk 1 above)
Comment (MichaIng) [reply dietpi/dietpi-software:1516]: @Fourdee 
Jep, it depends on the file ending as well, AFAIK. For .txt/.md files I remember a default of `<tab>` being translated into 2 times `<blank>`. But at least all our script files have tab -> tab with 8 blank width.

Ah now I see you format intention, having all `=` aligned.

**Comment by MichaIng** [comment]
🈴 Fails to start on Jessie. Verbose output: `Starting fahclient ... /usr/bin/FAHClient: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version 'GLIBCXX_3.4.21' not found (required by /usr/bin/FAHClient)`
- If we do not already apply a manual/backport update this particular library, I think we should drop Jessie support. Manually mess around with those library versions may otherwise break other software/system parts?

Ref: https://github.com/FoldingAtHome/fah-issues/issues/1147#issuecomment-134731211
- `libstdc++.so.6.0.21`+ required.
```
root@VM-Jessie:/tmp# l /usr/lib/x86_64-linux-gnu/ | grep libstdc++
lrwxrwxrwx 1 root root       19 Jan 31  2018 libstdc++.so.6 -> libstdc++.so.6.0.20
-rw-r--r-- 1 root root  1008120 Jan 31  2018 libstdc++.so.6.0.20
```
- 😝 one subversion too low.

**Comment by Fourdee** [comment]
@MichaIng 

> If we do not already apply a manual/backport update this particular library, I think we should drop Jessie support. Manually mess around with those library versions may otherwise break other software/system parts?

Yep. The only Jessie image required is for Odroid's, due to Kodi support. Aside from that, we should be not be implementing any Jessie specific workarounds or support.

**Comment by Fourdee** [comment]
Still work pending, but i've merged re: https://github.com/Fourdee/DietPi/issues/1985#issuecomment-410786747, and, testing purposes.

**Comment by MichaIng** [comment]
@Fourdee 
Okay, so drop Jessie support and I would go for "Computational Science" now, which seems to fit best: https://en.wikipedia.org/wiki/Computational_science
**€: Done**

Testing needed, it also on slower SBCs to 5 seconds wait time, before adjusting `config.xml` is enough. When the service stops, it writes back the parsed configuration to this file, which takes a short time, after the service stop is already finished (does not wait for it).
Only after the `config.xml` is written back, we can edit it, otherwise our edits will be overwritten.

**Comment by Fourdee** [comment]
@MichaIng 

If the config file does not exist until after 1st run is completed, you could try:
```
run_program

while true
do

    if [[ ! -f config.xml ]]; then

        sleep 2

    else
        
        stop program
        apply configuration changes
        break

    fi

done
```

**Comment by MichaIng** [comment]
@Fourdee 
If I observed it right, the config.xml always exist. After deb install, a sample config.xml file is in place. On first start, the client recognizes that/if the file is untouched sample or not, and only in first case it creates (at first internally) a simplified/formatted new config based on the system (core count, GPU etc.).
This configuration is then written back/overwriting the sample config.xml when service fully stopped.

If service was started with sample config.xml in place, and then G_CONFIG_INJECT applies, on service stop all manual changes will be overwritten. If G_CONFIG_INJECT applies, before service fully started, then fahclient does not adjust it any more and will fail, as needed values are missing/wrong.

But the current method (waiting for full start/stop) should work? As far as I tested, service names are always the same and match the pattern.

**Pull request merged**